src/sexpr.rs

```diff
@@ -93,37 +110,53 @@ struct ArenaInner {
     list_map: HashMap<&'static [SExpr], SExpr>,
 }
 
+impl ArenaInner {
+    pub fn intern_string(&mut self, s: impl Into<String> + AsRef<str>) -> u32 {
+        let ix = self.strings.len() as u32;
+
+        let s: String = s.into();
+
+        // // Safety argument: the name will live as long as the context as it is inserted into
+        // // the vector below and never removed or resized.
+        let s_ref: &'static str = unsafe { std::mem::transmute(s.as_str()) };
```

Comment:
I don't think the `AsRef<str>` bound is needed here because we immediately turn `s` into a `String` and never call `.as_ref()` on it.

Also, there is a weird double comment `// // ` thing going on.